src/silx/io/h5py_utils.py

```diff
@@ -88,17 +96,15 @@ def _hdf5_file_locking(mode="r", locking=None, swmr=None, libver=None, **_):
         * libver > v108 and file already locked: does not work
         * windows and HDF5_HAS_LOCKING_ARGUMENT and file already locked: does not work
 
-    :param str or None mode: read-only by default
-    :param bool or None locking: by default it is disabled for `mode='r'`
-                                 and `swmr=False` and enabled for all
-                                 other modes.
-    :param bool or None swmr: try both modes when `mode='r'` and `swmr=None`
-    :param None or str or tuple libver:
-    :returns bool:
+    :param mode: read-only by default
+    :param locking: by default it is disabled for `mode='r'`
+                    and `swmr=False` and enabled for all other modes.
```

Comment:
Perhaps say this:

"enabled for all other modes" -> "enabled when supported for all other modes"